### ISSUE TITLE
message drafts sometimes lost

### ISSUE THREAD

**Issue by honeymanfr** [signalapp/Signal-Android#1897]
Hi,
I experienced some frustration when starting a few messages, I had to check and copy some details to complete my messages.
Coming back to my uncompleted messages, each time, no draft was memorized so I had to start all over ...
Hope I am clear.
Did I miss something or do others share the same issue ?
Thank you.

Pierre Marie
Paris, France

**Comment by gloaming** [comment]
I have this issue on Moto G, Android 4.4.4

Steps to reproduce:
- Start a new message, or edit an existing draft
- Don't close the keyboard
- Kill TextSecure - Doesn't seem to matter how, but it has to actually die, not just disappear from the app switcher

Result: Draft is usually gone.
Expected: Draft should always be saved.

.

- You don't need to kill manually, you can switch out of the app, and wait for the system to do it
- If you do close the keyboard it will sometimes happen anyway
- It doesn't seem to happen if you back out to the thread list, though
- When re-opening TextSecure, the *previous* draft will display in the thread list if one existed, sometimes even when the new draft has been saved

**Comment by rhodey** [comment]
I'm able to reproduce this using the steps provided by @gloaming on my Nexus 5 running stock Android Lollipop 5.0.1 and TS 2.5.3. Looking into it now...

**Comment by t3io** [comment]
I've recently seen an instance of this bug and would like to offer some data in case it helps. I had one text draft and one audio draft in my conversations but then I had to switch off my phone as I was going through airport security. When I restarted I could see both drafts still there in the conversation list, but as soon as I selected the conversation with the audio draft, the draft disappeared and the message "there was an error setting your attachment" was displayed. Log: https://gist.github.com/4a5521aa2676c62f8dc4a4a60c21f37f

I then clicked on the conversation with the text draft, which made the draft also disappear. Log: https://gist.github.com/936ecfc7cd21c8c9aa880a8271a48b10

Hope these help debug.
